manual/core/integration/README.md

```diff
@@ -496,6 +497,18 @@ don't use geospatial types anywhere in your application, you can exclude the dep
 </dependency>
 ```
 
+Starting with driver 4.14.0 Esri has been changed to an optional dependency.  You no longer have to
```

Comment:
Now that we are doing this for Esri, I wonder why we wouldn't do the same for Jackson. Do you mind opening a ticket for that, if it doesn't exist yet?

Reply:
This isn't quite as clear cut for me.  Jackson is directly used at several spots within the driver:

* Insights code
* Graph code
* Cloud config code
* JsonCodec

The first is a no-op now that Insights is defunct; if anything I'd argue the driving code on our side should be removed, but that's a distraction from our current conversation.  The second usage above fits nicely into the model we're establishing here: dependencies of DSE features should be made optional.  If we stopped there I'd be more inclined to agree.

It's the last two that give me pause.  I don't know that we want to require users to manually add a dep when using the driver against Astra.  I'd make a similar argument for JsonCodec, although I readily agree it's less significant than the Astra use case.

Worth noting: the user who created the original PR to make ESRI optional did so because of the old Jackson version that came along with it.  Just making ESRI optional resolves that by using the more recent version we explicitly include.

Whaddya think?

Reply:
JsonCodec is optional, the driver doesn't register it by default, so I still think it falls under the "could-be-optional" category. Insights and Graph too.

It's only Astra that bugs me. 

After some more thought, OK to keep it mandatory. I'd make a case for creating a ticket to remove the dependency to Jackson in `CloudConfigFactory` first, and only then make Jackson optional.

> the user who created the original PR to make ESRI optional did so because of the old Jackson version that came along with it.

Indeed, Esri comes with two annoying dependencies because they are old and popular at the same time:

* org.json:json:20210307
* org.codehaus.jackson:jackson-core-asl:1.9.12

2 additional questions:

* Are we sure that the above dependencies are now absent from the shaded jar as well?
* Have you seen this? https://github.com/spring-projects/spring-boot/issues/29387#issuecomment-1015332594 In short: `org.json:json-api` was promoted to direct dependency for obscure OSGi-related reasons. Please check what's the current status of this dependency in the core pom.xml, in the shaded jar and in the OSGi tests.

Reply:
I naively thought that these would be removed by my earlier changes.  As I look back on it I'm not sure why I thought that but I did. :(  Regardless, this should be fixed now in the most recent push.

I'll look into the spring-boot issue as well.

Reply:
I naively thought that these would be removed by my earlier changes.  As I look back on it I'm not sure why I thought that but I did. :(  Regardless, this should be fixed now in the most recent push.

I'll look into the spring-boot issue as well.

Reply:
I can also confirm that making ESRI optional has the effect of making it's transitive dependencies optional.  That means that org.json:json is now an optional dependency for core and core-shaded.  I'll add a comment onto that Spring ticket confirming that fact.